src/hooks/useColumns.tsx

```diff
@@ -0,0 +1,160 @@
+import { Tag } from 'antd';
+import {
+  ColumnsType,
+  FilterValue,
+  SorterResult,
+} from 'antd/es/table/interface';
+import { useContext, useMemo } from 'react';
+import { Film } from '../types/Film';
+import { Genre } from '../types/Genre';
+import { genreColor } from '../utils/genreColor';
+import { getTranslation } from '../utils/getTranslation';
+import { LangContext } from '../utils/LangContext';
+
+interface Params {
+  posterURL: string;
+  sortedInfo: SorterResult<Film>;
+  genres: Genre[];
+  filteredInfo: Record<string, FilterValue | null>;
+}
+
+export const useFilmColumns = ({
+  sortedInfo,
+  posterURL,
+  genres,
+  filteredInfo,
+}: Params) => {
+  const lang = useContext(LangContext);
+
+  const columns: ColumnsType<Film> = useMemo(
+    () => [
+      {
+        title: getTranslation('table.poster', lang),
+        dataIndex: 'poster_path',
+        key: 'poster',
+        align: 'center',
+        render: (imgUrl: string) => (
+          <div>
+            <img
+              src={posterURL + imgUrl}
+              alt="poster"
+              style={{
+                width: '100%',
+                objectFit: 'cover',
+              }}
+            />
+          </div>
+        ),
```

Comment:
it's not the best solution to return the JSX from hooks. maybe you can create a separate component for render and use it here?
```suggestion
        render: PosterImage,
```